addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -412,17 +412,18 @@ private void processApplicationStateChanged(String method, JsonObject json) {
 
             this.volume = volume;
         } else {
-            logger.debug("Unknown event from kodi {}: {}", method, json.toString());
+            logger.debug("Unknown event from Kodi {}: {}", method, json.toString());
         }
+        listener.updateConnectionState(true);
     }
 
     private void processSystemStateChanged(String method, JsonObject json) {
-        if ("System.OnQuit".equals(method) || "System.OnSleep".equals(method) || "System.OnRestart".equals(method)) {
-            try {
-                listener.updateConnectionState(false);
-            } catch (Exception e) {
-                logger.error("Event listener invoking error", e);
-            }
+        if ("System.OnQuit".equals(method) || "System.OnRestart".equals(method) || "System.OnSleep".equals(method)) {
+            listener.updateConnectionState(false);
```

Comment:
@cweitkamp is it somehow likely that the catch served a purpose, will things break now in certain cases. Good to remove it but did you consider which potential problems where hidden here?